File: library-java/src/main/java/com/libraryjava/domain/user/loanhistory/UserLoanHistory.java
package com.libraryjava.domain.user.loanhistory;

import com.libraryjava.domain.user.User;
import com.libraryjava.domain.user.UserStatus;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@Getter
public class UserLoanHistory {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String bookName;

    @Enumerated(EnumType.STRING)
    private UserLoanStatus userLoanStatus;

    @ManyToOne(fetch = FetchType.LAZY)
    private User user;

    public UserLoanHistory(String bookName, UserLoanStatus userLoanStatus, User user) {
        this.bookName = bookName;
        this.userLoanStatus = userLoanStatus;
        this.user = user;
    }

    public void doReturn() {
        this.userLoanStatus = UserLoanStatus.RETURNED;
    }
}

File: src/main/kotlin/com/library/domain/user/loanhistory/UserLoanHistory.kt
package com.library.domain.user.loanhistory

import com.library.domain.user.User
import jakarta.persistence.*

@Entity
class UserLoanHistory(
    @ManyToOne
    val user: User,

    val bookName: String,

    @Enumerated(EnumType.STRING)
    var status: UserLoanStatus = UserLoanStatus.LOANED,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
) {

    fun doReturn() {
        this.status = UserLoanStatus.RETURNED
    }

    companion object{
        fun fixture(
            user: User,
            bookName: String = "이상한 나라의 앨리스",
            status: UserLoanStatus = UserLoanStatus.LOANED,
        ): UserLoanHistory {
            return UserLoanHistory(
                user = user,
                bookName = bookName,
                status = status,
            )
        }
    }
}